aspnetcore/razor-components/components.md

```diff
@@ -322,6 +322,53 @@ It's often convenient to close over additional values, such as when iterating ov
 > [!NOTE]
 > Do **not** use the loop variable (`i`) in a `for` loop directly in a lambda expression. Otherwise the same variable is used by all lambda expressions causing `i`'s value to be the same in all lambdas. Always capture its value in a local variable (`buttonNumber` in the preceding example) and then use it.
 
+### EventCallback
+
+`EventCallback` is used to dispatch event routing between components. An `EventCallback` is a value type that wraps a delegate with a reference to the component that created it.
+
+The sample app demonstrates how a button's `onclick` handler receives an `EventCallback<UIMouseEventArgs>` delegate.
+
+*Child component*:
+
+[!code-cshtml[](common/samples/3.x/BlazorSample/Pages/ChildComponent.cshtml?highlight=5-7,17-18)]
+
+The Parent component sets the child's `EventCallback<T>` to its `ShowMessageTriggeredByChild` method.
+
+*Parent component*:
+
+[!code-cshtml[](common/samples/3.x/BlazorSample/Pages/ParentComponent.cshtml?name=snippet_ParentComponent&highlight=6,16-19)]
+
+When the button is selected in the Child component:
+
+* The Parent component's `ShowMessageTriggeredByChild` method is called.
+* `messageText` is updated in the Parent component.
+* The Parent component is rerendered, and the `messageText` value is displayed.
+
+`EventCallback` and `EventCallback<T>` permit asynchronous delegates. `EventCallback<T>` is strongly typed and requires a specific argument type. `EventCallback` is weakly typed and allows any argument type.
+
+```chstml
+<p><b>@messageText</b></p>
+
+@{ var message = "Default Text"; }
+
+<ChildComponent 
+    OnClick="@(async () => { await Task.Yield(); messageText = "Blaze It!"; }" />
+
+@function {
+    string messageText;
+}
+```
+
+Invoke an `EventCallback` or `EventCallback<T>` with `InvokeAsync` and await the <xref:System.Threading.Tasks.Task>:
+
+```csharp
+await callback.InvokeAsync(arg);
+```
+
+Use `EventCallback` and `EventCallback<T>` for event handling and binding component parameters. Don't use `EventCallback` and `EventCallback<T>` for child content&mdash;continue to use `RenderFragment` and `RenderFragment<T>` for child content.
+
+Prefer the strongly typed `EventCallback<T>`, which provides better error feedback to users of the component. Use `EventCallback` when there's no value passed to the callback.
```

Comment:
I believe another benefit of EventCallback is that specifying the event parameter in your event callback is optional, just like it is for other UI event handlers.